internal/replicationslot/repo/repo_test.go

```diff
@@ -0,0 +1,71 @@
+package repo_test
+
+import (
+	"context"
+	"database/sql"
+	"testing"
+	"time"
+
+	_ "github.com/jackc/pgx/v5/stdlib" // PostgreSQL driver
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/postgres"
+	"github.com/testcontainers/testcontainers-go/wait"
+	replicationSlotRepo "reactive-tech.io/kubegres/internal/replicationslot/repo"
+)
+
+func TestReplicationSlotsWithTestcontainers(t *testing.T) {
+	// This test requires Docker to be running.
+	if testing.Short() {
+		t.Skip("skipping testcontainers test in short mode")
+	}
+
+	pgContainer, err := postgres.Run(t.Context(),
+		"postgres:latest",
```

Comment:
We should use the same default version that is somehow hardcoded, maybe in the e2e test.